lib/phoenix_html/tag.ex

```diff
@@ -311,8 +311,32 @@ defmodule Phoenix.HTML.Tag do
       img_tag(Routes.static_path(@conn, "/logo.png"))
       <img src="/logo-123456.png?vsn=d">
 
+  For responsive images, pass a map, list or string through `:srcset`.
+
+      img_tag("/logo.png", srcset: %{"/logo.png" => "1x", "/logo-2x.png" => "2x"})
+      <img src="/logo.png" srcset="/logo.png 1x, /logo-2x.png 2x">
+
+      img_tag("/logo.png", srcset: ["/logo.png", {"/logo-2x.png", "2x"}])
+      <img src="/logo.png" srcset="/logo.png, /logo-2x.png 2x">
+
   """
   def img_tag(src, opts \\ []) do
+    opts =
+      case Keyword.pop(opts, :srcset) do
+        {nil, opts} -> opts
+        {srcset, opts} -> Keyword.put(opts, :srcset, stringify_srcset(srcset))
```

Comment:
```suggestion
        {srcset, opts} -> [srcset: stringify_srcset(srcset)] ++ opts
```

Reply:
No need to go through `put` since we have popped.

Reply:
Sure, fixed. There’s a `put` after a `pop` at line 240 in case you want to refactor.

Reply:
Sure, fixed.